internal/kafka/internal/api/dbapi/kafka_request_types.go

```diff
@@ -141,3 +143,67 @@ func (k *KafkaRequest) GetExpirationTime(lifespanSeconds int) *time.Time {
 	expireTime := k.CreatedAt.Add(time.Duration(lifespanSeconds) * time.Second)
 	return &expireTime
 }
+
+// The RemainingLifeSpan is the remaining life a kafka instance have before it expires and gets deleted
+type RemainingLifeSpan interface {
+	// Infinite returns whether the Kafka instance has infinite (no expiration) remaining life span
+	Infinite() bool
+
+	// Days returns the number of remaining life span days a Kafka instance has. If infinite, returns -1
+	Days() float64
+}
+
+var _ RemainingLifeSpan = &remainingLifeSpan{}
+
+// remainingLifeSpan is an implementation of the RemainingLifeSpan interface
+type remainingLifeSpan struct {
+	// remainingLifeSpanDays is the number of days before the Kafka instance will be deleted. It is a `float64` to be able to manage fraction of days.
+	remainingLifeSpanDays float64
+	// infinite represent whether the Kafka instance has an expiration
+	infinite bool
+}
+
+func (l *remainingLifeSpan) Infinite() bool {
+	return l.infinite
+}
+
+func (l *remainingLifeSpan) Days() float64 {
+	return l.remainingLifeSpanDays
+}
+
+// IsExpired returns whether a kafka is expired and how many days the instance will live before expiring (-1 never expires)
+func (k *KafkaRequest) IsExpired() (bool, RemainingLifeSpan) {
+	if !k.ExpiresAt.Valid {
+		return false, &remainingLifeSpan{
+			remainingLifeSpanDays: -1,
+			infinite:              true,
+		}
+	}
+
+	now := time.Now()
+	if now.After(k.ExpiresAt.Time) {
+		return true, &remainingLifeSpan{
+			remainingLifeSpanDays: 0,
+			infinite:              false,
+		}
+	}
+
+	remainingLife := k.ExpiresAt.Time.Sub(now)
```

Comment:
How does `Sub` behaves if expires_at is earlier than now?

For example:
What happens if it expires 20 october and now is 22 october
What happens if it expires expires 20 october and now is 10 october

Reply:
Returns a negative result: https://go.dev/play/p/I1oUTnIDqep

Reply:
Then what happens if it goes late by 1 day? The result would be -1 which is considered infinite :thinking:

Reply:
-1 is not considered infinite. Infinite is when `Infinite` returns `true` and that happens only when `ExpiredAt` is `nil`

Reply:
In the method documentation the following is said:
```
	// Days returns the number of remaining life span days a Kafka instance has. If infinite, returns -1
```

It is not considered infinite but -1 is returned if it is infinite.
Maybe we could add that it also can return negative number when it is not infinite meaning the days after the expiration that have elapsed?

Reply:
Documentation updated!

Reply:
:+1: